business_objects/record_label_association.py

```diff
@@ -342,6 +364,28 @@ def create_token_objects(
     ]
 
 
+def create_by_record_user_label_dict(
+    project_id: str, record_user_label_dict: Dict[str, Any], with_commit: bool = False
+):
+    create_list = []
+    for record_id, user_dict in record_user_label_dict.items():
+        for user_id, tasks_dict in user_dict.items():
+
+            rlas = [
+                RecordLabelAssociation(
+                    project_id=project_id,
+                    record_id=record_id,
+                    labeling_task_label_id=label_id,
+                    source_type=enums.LabelSource.MANUAL.value,
+                    return_type=enums.InformationSourceReturnType.RETURN.value,
+                    created_by=user_id,
+                )
+                for label_id in tasks_dict.values()
+            ]
+            create_list.extend(rlas)
```

Comment:
extending a list often will run into performance issues, especially once they reach a certain size since space needs to be allocated beforehand and then copied from a to b.

if think if we chunk the data beforehand it's not super dramatic however with currently unlimited amounts of data not optimal

Reply:
understood. Since it is now done in chunks i would leave it as it is but complelty valid point